bin/rule.ml

```diff
@@ -117,7 +117,11 @@ let run () md_file section direction prelude prelude_str root =
   let on_item acc = function
     | Mdx.Section _ | Text _ -> acc
     | Block b when active b ->
-      let files, dirs, nd = acc in
+      let files, dirs, nd, requires = acc in
+      let requires =
+        Mdx.Block.required_packages b
+        |> List.fold_left (fun s e -> String.Set.add e s) requires
```

Comment:
This is `String.Set.of_list`!

Reply:
This adds to `requires` so it's `of_list` and `union`. I think this is better.

Reply:
The fact that it's going to be refactored later doesn't mean we shouldn't properly name variables now. It's rather the opposite actually, making the code easier to understand helps with later refactoring.

I don't mind if you leave it like that as it is in a way unrelated to this PR. I just think it's good practice to improve the code we're modifying.

Reply:
Do you mean it's better performance-wise or it reads better?

I'm not sure it reads really well, in particular:
```ocaml
let requires_from_block = String.Set.of_list (Mdx.Block.required_packages b) in
let requires = String.Set.union requires requires_from_block in
...
```
would be much clearer than the current where I have to mentally parse the fold left.

Anyway don't bother with that as I think that while rebasing #150 on top of this, I refactored it that way because it introduce different sources for required packages and it made sense to extract intermediate variables there.